### PR TITLE
Grisha

### PR DESCRIPTION
**Pull request by grysha11**
Done with the parsing:
Flood fill algorithm,
Tab triming,
Colors and textures validation,
Refactored everything to the separate files,
Norminette done.

### CODE REVIEW

**Comment by tamasandor** [comment]
We have still reachable with this map: 
``` 
NO			inc/north.xpm

EA		inc/east.xpm







SO	inc/purplestone.xpm




WE	                            inc/west.xpm

C 0,0,0



F 0,0,898

111111111111111
111111000111111
111110000011111
1111000S0001111
111000000000111
110000000000011
100001111100001
110000111000011
111000010000111
111100000001111
111110000011111
111111000111111
111111111111111

```

Looks like an easy fix

**Comment by tamasandor** [comment]
> We have still reachable with this map:
> 
> ```
> NO			inc/north.xpm
> 
> EA		inc/east.xpm
> 
> 
> 
> 
> 
> 
> 
> SO	inc/purplestone.xpm
> 
> 
> 
> 
> WE	                            inc/west.xpm
> 
> C 0,0,0
> 
> 
> 
> F 0,0,898
> 
> 111111111111111
> 111111000111111
> 111110000011111
> 1111000S0001111
> 111000000000111
> 110000000000011
> 100001111100001
> 110000111000011
> 111000010000111
> 111100000001111
> 111110000011111
> 111111000111111
> 111111111111111
> ```
> 
> Looks like an easy fix

Same with minus rgb values. 
If the number is not `1` or `0` in the map content my code breaks. I could handle it in my part but I think this should throw a map error.
```

111111111111111
111111000111111
111110000011111
1111000S0001111
111000000000111
110000000000011
100001111100001
110000111900011
111000010000111
111100000001111
111110000011111
111111000111111
111111111111111

```

**Comment by tamasandor** [comment]
Map is not closed error also creates still reachable:

```
NO			inc/north.xpm

EA		inc/east.xpm







SO	inc/purplestone.xpm




WE	                            inc/west.xpm

C 0,0,0



F 0,5,0

111111111111111
111111000111111
111110000011111
1111000S0001111
111000000000111
110000000000011
100001111100001
110000111000011
111000010000111
111100000001111
111110000011111
111111000111101000000
111111111111111

```

**Comment by tamasandor** [comment]
Valgrind error on this map (Invalid read of size 1)

```
111111111111111
111111000111111
111110000011111
1111000S0001111
111000000000111
110000000000011
100001111100001
110000111000011
111000010000111
111100000001111
111110000011111
111111000111111111000001111
111111111111111
```

**Comment by grysha11** [comment]
> > We have still reachable with this map:
> > ```
> > NO			inc/north.xpm
> > 
> > EA		inc/east.xpm
> > 
> > 
> > 
> > 
> > 
> > 
> > 
> > SO	inc/purplestone.xpm
> > 
> > 
> > 
> > 
> > WE	                            inc/west.xpm
> > 
> > C 0,0,0
> > 
> > 
> > 
> > F 0,0,898
> > 
> > 111111111111111
> > 111111000111111
> > 111110000011111
> > 1111000S0001111
> > 111000000000111
> > 110000000000011
> > 100001111100001
> > 110000111000011
> > 111000010000111
> > 111100000001111
> > 111110000011111
> > 111111000111111
> > 111111111111111
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Looks like an easy fix
> 
> Same with minus rgb values. If the number is not `1` or `0` in the map content my code breaks. I could handle it in my part but I think this should throw a map error.
> 
> ```
> 
> 111111111111111
> 111111000111111
> 111110000011111
> 1111000S0001111
> 111000000000111
> 110000000000011
> 100001111100001
> 110000111900011
> 111000010000111
> 111100000001111
> 111110000011111
> 111111000111111
> 111111111111111
> ```

I forgor to make this check and also player count check

**Comment by grysha11** [comment]
Fixed all of the errors, waiting for your double check.

**Comment by tamasandor** [comment]
Now this one is creating definitely lost (when there is extra space between the texture full path)
```
NO			inc/north.xpm

EA		inc/east.xpm







SO	inc/purplestone.xpm




WE	                            inc / west.xpm

C 0,0,0



F 0,0,0

111111111111111
111111000111111
111110000011111
1111000S0001111
111000000000111         111111
110000000000011        11100011
100001111100001         11111
110000111000011
111000010000111
111100000001111
111110000011111
111111000101111
111111111111111
```

**Comment by grysha11** [comment]
Fixed issue with free_parse() function

File: cub3d.h [hunk 1]
```diff
@@ -20,6 +20,9 @@
 # include "./minilibx/mlx.h"
 # include "inc/gnl/gnl.h"
 # include "inc/libft/libft.h"
+# include "./minilibx/mlx.h"
+# include "inc/gnl/gnl.h"
+# include "inc/libft/libft.h"
```
Comment (tamasandor) [inline cub3d.h]: We had it here already :D

File: main.c [hunk 2]
```diff
@@ -98,12 +114,12 @@ int	main(int ac, char **av)
...
 	init_parse(ac, av, parse);
 	set_up_player(&mlx);
-	print_parse(parse);
 	if (mlx_setup(&mlx))
 		return (1);
 	set_textures(&mlx, parse->textures);
 	mlx_loop_hook(mlx.mlx, render, &mlx);
 	mlx_loop(mlx.mlx);
 	clean_exit(&mlx);
+	free_parse(parse);
```
Comment (tamasandor) [inline main.c]: Isn't clean exit calling the free_parse already?

**Review by tamasandor — APPROVED** [review]
Nice work, thank you for fixing the issues that I pointed out!

**Pull request merged**